packages/zoe/src/types.js

```diff
@@ -0,0 +1,218 @@
+ * is completed, the promise for each payment resolves after the remote
+ * issuer successfully withdraws the payment.
+ *
+ * @property {Promise<OfferOutcome>} outcome Note that if the offerHook throws,
+ * this outcome Promise will reject, but the rest of the OfferResultRecord is
+ * still meaningful.
+ *
+ * @property {CompleteObj} [completeObj]
+ * completeObj will only be present if exitKind was 'onDemand'
+ *
+ * @typedef {{give?:AmountKeywordRecord,want?:AmountKeywordRecord,exit?:ExitRule}} Proposal
+ *
+ * @typedef {Object.<string,Amount>} AmountKeywordRecord
+ *
+ * The keys are keywords, and the values are amounts. For example:
+ * { Asset: amountMath.make(5), Price: amountMath.make(9) }
+ *
+ * @typedef {AmountKeywordRecord[]} AmountKeywordRecords
+ *
+ * @typedef {Object} MakeInstanceResult
+ * @property {Promise<Invite>} invite
+ * @property {InstanceRecord} instanceRecord
+ */
+
+/**
+ * @typedef {Object} Timer
+ * @typedef {number} Deadline
+ *
+ * @typedef {{waived:null}} Waived
+ * @typedef {{onDemand:null}} OnDemand
+ *
+ * @typedef {{afterDeadline:{timer:Timer, deadline:Deadline}}} AfterDeadline
+ *
+ * @typedef {(Waived|OnDemand|AfterDeadline)} ExitRule
+ * The possible keys are 'waived', 'onDemand', and 'afterDeadline'.
+ * `timer` and `deadline` only are used for the `afterDeadline` key.
+ * The possible records are:
+ * `{ waived: null }`
+ * `{ onDemand: null }`
+ * `{ afterDeadline: { timer :Timer<Deadline>, deadline :Deadline } }
+ */
+
+/**
+ * @typedef {Object} Invite
+ * An invitation to participate in a Zoe contract.
+ * Invites are Payments, so they can be transferred, stored in Purses, and
+ * verified. Only Zoe can create new Invites.
+ * @property {() => Brand} getAllegedBrand
+ */
+
+/**
+ * @callback MakeContract The type exported from a Zoe contract
+ * @param {ContractFacet} zcf The Zoe Contract Facet
+ * @returns {Invite} invite The closely-held administrative invite
+ */
+
+/**
+ * @typedef {{}} InstanceHandle - an opaque handle for a contract instance
+ * @typedef {{}} OfferHandle - an opaque handle for an offer
+ * @typedef {{}} InviteHandle - an opaque handle for an invite
+ * @typedef {Object} CustomProperties
+ *
+ * @typedef {object} OfferRecord
+ * @property {OfferHandle=} handle - opaque identifier, used as the table key
+ * @property {InstanceHandle} instanceHandle - opaque identifier for the instance
+ * @property {Proposal} proposal - the offer proposal (including want, give, exit)
+ * @property {Allocation} currentAllocation - the allocation corresponding to this offer
+ * @property {Notifier<any>} notifier - the notifier for XXX
+ *
+ * @typedef {object} InstanceRecord
+ * @property {InstanceHandle=} handle - opaque identifier for the instance, used as the table key
+ * @property {InstallationHandle} installationHandle - opaque identifier for the installation
+ * @property {Promise<Object.<string,function>>} publicAPI - the invite-free publicly accessible API for the contract
```

Comment:
I think Chris is thinking this won't be a promise

Reply:
@Chris-Hibbert I've changed this just to `Object<string,function>`, and hope your code will mesh with this definition by the time you merge to master.

Reply:
Not anything to worry about: my PR won't pass tests until it meshes.